### PR TITLE
Prepare for use

### PR DESCRIPTION
**Pull request by tehpsalmist**
This merges in significant changes to the structure of the repository, converting it to a Lerna Monorepo, so each package (server, client, generator) can be individually published and pulled for the various projects that need them from npm.

[Lerna Documentation](https://lerna.js.org/)

Three minor things to note:

1. Coding style (format) throughout the repository was inconsistent and failing linting by eslint, so changes were made to conform everything to mirror Mayan, but if that's a problem later, formatting changes can be easily made and committed.
2. The concept of targeting "dev" for logging purposes was removed entirely, in favor of allowing the using developer to toggle logging using the `logging()` method.
3. Tests are passing, but the example project is still broken. This seemed insigificant, due to the fact that this project is receiving more relevant testing in a variety of environments at this point, but perhaps we can get that example fixed in the future for the benefit of the open source community, if it comes to that.

### CODE REVIEW

**Comment by evert0n** [comment]
@tehpsalmist would be good to add a simple README for both packages see in npm 
- https://www.npmjs.com/package/@zenginehq/post-rpc-client
- https://www.npmjs.com/package/@zenginehq/post-rpc-server

**Comment by wizehive-seanburke** [comment]
It may be a good idea to add a .editorconfig file into the repository as well.

File: package.json [hunk 1]
```diff
@@ -5,51 +5,42 @@
...
+    "build": "lerna run build",
+    "build-dev": "lerna run build-dev",
     "clean": "rm -f lib/*",
-    "example": "webpack-dev-server --content-base example/ --host localhost --port 5005 --colors --watch",
+    "example": "webpack-dev-server --content-base ./example --host localhost --port 5005 --env dev --colors --watch --config webpack.example.config.js",
     "blanket": "webpack-dev-server --host localhost --port 6001 --colors --watch",
-    "test": "webpack --env prod && grunt karma:prod"
-  },
-  "dependencies": {
-    "native-promise-only": "^0.8.1",
```
Comment (wizehive-seanburke) [inline package.json]: This is still being used in client.js

Comment (tehpsalmist) [reply package.json]: @wizehive-seanburke what are you running to build it? the command should be `lerna run build`. I will work on documentation tomorrow, to address this, and what @evert0n pointed out above. Thanks for the feedback, y'all.

Comment (wizehive-seanburke) [reply package.json]: The error was from running that command in the base folder.

I was changing up node versions with nvm so I removed the node_modules folder, did a `npm install` then ran `npm build`, which runs `lerna run build`. During the build it started complaining about that dep missing.

Comment (tehpsalmist) [reply package.json]: you need to run lerna bootstrap, I believe.

Comment (wizehive-seanburke) [reply package.json]: That was it, I assumed it would automatically check and do anything like that on a build run due to the lerna.json file.

Comment (tehpsalmist) [reply package.json]: There's a little _lerna-ing_ curve, eh? *slaps knee...

**Pull request merged**